exes/mqtt-broadcaster/inc/config/mqtt_config.hpp

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <cstdint>
+#include <string>
+#include <variant>
+
+#include <glaze/glaze.hpp>
+#include <tfc/confman.hpp>
+
+namespace tfc::mqtt {
+
+enum struct port_e : std::uint16_t { mqtt = 1883, mqtts = 8883 };
+
+enum struct ssl { yes, no };
+
+// File under /etc/tfc/mqtt-broadcaster/def/mqtt_broadcaster.json which specifies the connection values for the MQTT broker.
+struct broker_config {
+  std::string address{};
+  std::variant<port_e, uint16_t> port{};
+  ssl ssl_active{};
+  std::string username{};
+  std::string password{};
+  std::string client_id{};
+
+  struct glaze {
+    // clang-format off
+    static constexpr auto value{ glz::object(
+        "address", &broker_config::address, "Hostname or IP address of the MQTT broker",
+        "port", &broker_config::port, "Port of the MQTT broker. Possible values are: mqtt, mqtts or a custom port number",
+        "ssl_active", &broker_config::ssl_active, "Whether or not to use SSL to connect to the MQTT broker",
+        "username", &broker_config::username, "Username for the MQTT broker",
+        "password", &broker_config::password, "Password for the MQTT broker",
+        "client_id", &broker_config::client_id, "Spark Plug B Client ID, used to identify which client is sending information"
+      ) };
+    // clang-format on
+    static constexpr auto name{ "mqtt_broadcaster" };
```

Comment:
PLEASE do not reuse `mqtt_broadcaster` in struct names. It should be a compile error, as in not allowed.
It is here and in couple of other places